### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by Ao-senXiong**
When I updating the codebase to opprop, I found some typos. Then I used intellij spellcheck and inspect all our codebase to found more typos and fixes them.

### CODE REVIEW

File: checker/src/main/java/org/checkerframework/checker/regex/RegexVisitor.java [hunk 1]
```diff
@@ -105,7 +105,8 @@ public Void visitMethodInvocation(MethodInvocationTree tree, Void p) {
                     // When checking implementations of java.util.regex.MatcherResult, calls to
                     // group (and other methods) don't have a receiver tree.  So, just do the
                     // regular checking.
-                    // Verifying an implemenation of a subclass of MatcherResult is out of the scope
+                    // Verifying an implementation of a subclass of MatcherResult is out of the
+                    // scope
                     // of this checker.
```
Comment (wmdietl) [inline checker/src/main/java/org/checkerframework/checker/regex/RegexVisitor.java:110]:
```suggestion
                    // scope of this checker.
```

Comment (wmdietl) [reply checker/src/main/java/org/checkerframework/checker/regex/RegexVisitor.java:110]: Why did the linebreak change here? Didn't spotless do this correctly?

File: framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenesStorage.java [hunk 2]
```diff
@@ -71,7 +71,7 @@
  *
  * <p>The set of annotations inferred for a certain class is stored in an {@link
  * org.checkerframework.afu.scenelib.annotations.el.AScene}, which {@code writeScenes()} can write
- * into a file. For example, a class {@code my.pakkage.MyClass} will have its members' inferred
+ * into a file. For example, a class {@code my.package.MyClass} will have its members' inferred
```
Comment (wmdietl) [inline framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenesStorage.java:74]: `package` is a keyword, so it cannot be used as a package name. So I think here `pakkage` is an intentional misspelling to be a valid package name.

File: framework/src/main/java/org/checkerframework/framework/util/CheckerMain.java [hunk 3]
```diff
@@ -683,7 +683,7 @@ public static String findPathTo(Class<?> cls, boolean errIfFromDirectory)
         }
         String name = cls.getName();
         String classFileName;
-        /* name is something like pakkage.name.ContainingClass$ClassName. We need to turn this into ContainingClass$ClassName.class. */
+        /* name is something like package.name.ContainingClass$ClassName. We need to turn this into ContainingClass$ClassName.class. */
```
Comment (wmdietl) [inline framework/src/main/java/org/checkerframework/framework/util/CheckerMain.java:686]: Same comment about `package` being a keyword.

File: framework/src/main/java/org/checkerframework/framework/util/TypeArgumentMapper.java [hunk 4]
```diff
@@ -265,7 +265,7 @@ private static List<TypeRecord> depthFirstSearchForSupertype(
     }
 
     /**
-     * Computes one level for depthFirstSearchForSupertype then recurses.
+     * Computes one level for depthFirstSearchForSupertype then recurse.
```
Comment (wmdietl) [inline framework/src/main/java/org/checkerframework/framework/util/TypeArgumentMapper.java:268]: I think the original was correct and consistent with the `Computes`.
You could reformulate the whole sentence to make it clearer.

File: framework/src/main/java/org/checkerframework/framework/util/defaults/QualifierDefaults.java [hunk 5]
```diff
@@ -567,7 +567,7 @@ private void applyDefaults(Tree tree, AnnotatedTypeMirror type) {
             case IDENTIFIER:
                 elt = TreeUtils.elementFromUse((IdentifierTree) tree);
                 if (ElementUtils.isTypeDeclaration(elt)) {
-                    // If the Idenitifer is a type, then use the scope of the tree.
+                    // If the Identifier is a type, then use the scope of the tree.
```
Comment (wmdietl) [inline framework/src/main/java/org/checkerframework/framework/util/defaults/QualifierDefaults.java:570]:
```suggestion
                    // If the identifier is a type, then use the scope of the tree.
```

File: framework/src/main/java/org/checkerframework/framework/util/typeinference/DefaultTypeArgumentInference.java [hunk 6]
```diff
@@ -443,7 +443,7 @@ private Map<TypeVariable, AnnotatedTypeMirror> infer(
      * }</pre>
      *
      * TODO: NOTE WE ONLY DO THIS FOR InferredType results for now but we should probably include
-     * targest as well
+     * target as well
```
Comment (wmdietl) [inline framework/src/main/java/org/checkerframework/framework/util/typeinference/DefaultTypeArgumentInference.java:446]: What does `target` refer to here? Maybe it should have been `targets`?

**Review by wmdietl — CHANGES_REQUESTED** [review]
Thanks for this! Just a few small comments.
I would like to first get #444 merged. That will likely conflict with a few of your fixes, sorry.

**Comment by Ao-senXiong** [comment]
> 

No problem, I am still testing the new version of initialization checker and I do not mind make some changes after it get merged in this PR.

**Comment by Ao-senXiong** [comment]
> Thanks for this! Just a few small comments. I would like to first get #444 merged. That will likely conflict with a few of your fixes, sorry.

No problem, I am still testing the new version of initialization checker and I do not mind make some changes after it get merged in this PR.

File: framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenesStorage.java (hunk 2 above)
Comment (Ao-senXiong) [reply framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenesStorage.java:74]: ok, I will change it back.

File: framework/src/main/java/org/checkerframework/framework/util/typeinference/DefaultTypeArgumentInference.java (hunk 6 above)
Comment (Ao-senXiong) [reply framework/src/main/java/org/checkerframework/framework/util/typeinference/DefaultTypeArgumentInference.java:446]: Done

**Review by wmdietl — APPROVED** [review]
Thanks!

**Pull request merged**